resources/install.rb

```diff
@@ -131,10 +153,22 @@
         action :create
       end
 
-      windows_zipfile ENV['ProgramW6432'] do
-        source "#{Chef::Config[:file_cache_path]}/#{file_name}"
-        not_if { ::File.exist?("#{ENV['ProgramW6432']}\\telegraf\\telegraf.exe") }
-        action :unzip
+      version_path = "#{ENV['ProgramW6432']}\\telegraf\\#{new_resource.install_version}"
+
+      unless ::File.exist?(::File.join(version_path, "telegraf", "telegraf.exe"))
```

Comment:
Maybe have a `executable_exists?` function in the helpers library so you aren't duplicating how paths are built?